### PR TITLE
Easier extendablity with custom commands.

### PR DESCRIPTION
**Pull request by yggdr**
This adds the ability for the user to add custom commands to a Monitor subclass, inspired by the standard library module "Cmd".

### CODE REVIEW

**Comment by jettify** [comment]
You mean https://docs.python.org/3/library/cmd.html this one? I did not know that such thing even exists :)

**Comment by jettify** [comment]
Sorry for delay, I will PR more closely in few days.

File: tests/test_monitor.py [hunk 1]
```diff
@@ -94,20 +108,29 @@ def test_basic_monitor(monitor, tn_client, loop):
     assert 'Task' in resp
 
     resp = execute(tn, 'ps 123\n')
-    assert 'Task' in resp
+    assert 'TypeError' in resp
```
Comment (jettify) [inline tests/test_monitor.py]: Can you elaborate on this one? Why this change?

File: tests/test_monitor.py [hunk 2]
```diff
@@ -94,20 +108,29 @@ def test_basic_monitor(monitor, tn_client, loop):
...
 
     resp = execute(tn, 'signal name\n')
     assert 'Unknown signal name' in resp
 
     resp = execute(tn, 'stacktrace\n')
     assert 'loop.run_forever()' in resp
 
-    resp = execute(tn, 'wehere 123\n')
```
Comment (jettify) [inline tests/test_monitor.py]: 🤦‍♂️ why this event worked?

Comment (yggdr) [reply tests/test_monitor.py]: Why it worked in the past you mean? Because with the old code any command beginning with a "w" would be routed to "where".

File: tests/test_monitor.py (hunk 1 above)
Comment (yggdr) [reply tests/test_monitor.py]: Because the new code for looking up the method to execute will take the method (`do_ps` in this example) and give it any arguments supplied by the user (`123`). The TypeError is simply the error that Python raises when you give an incorrect amount of arguments to a function.

I am currently working on implementing more of the public interface of Cmd where this will picked up.

**Comment by yggdr** [comment]
@jettify Yes, that's the standard library cmd module. I thought it useful to try and adhere to its API while making aiomonitor more extensible. If you have any comments about the direction this is taking, I'm listening :)

Building on this PR I'm working on adding more of the Cmd-API (like `precmd`, `postcmd`) to aiomonitor.

File: tests/test_monitor.py (hunk 2 above)
Comment (jettify) [reply tests/test_monitor.py]: yeah, stupid error on my side :)

**Comment by jettify** [comment]
I like your idea and implementation (thanks for updating docs and example!), give me few days to reread PR more carefully.

File: tests/test_monitor.py (hunk 2 above)
Comment (yggdr) [reply tests/test_monitor.py]: It did take me a while to realise what was going on as well :D
